lib/src/media/image.dart

```diff
@@ -0,0 +1,188 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:image/image.dart' as img;
+
+class MimeType {
+  final String _type;
+  final String _name;
+  static final Map<String, MimeType> _map = {
+    'image/vnd.viam.rgba': MimeType.viamRgba,
+    'image/jpeg': MimeType.jpeg,
+    'image/png': MimeType.png,
+    'pointcloud/pcd': MimeType.pcd,
+  };
+
+  /// The name of the MimeType, e.g. 'image/jpeg'
+  /// If the MimeType is not supported, then this [name] is the string of the unsupported MimeType.
+  String get name => _name;
+
+  const MimeType._(this._type, this._name);
+
+  static MimeType get viamRgba => MimeType._('viamRgba', 'image/vnd.viam.rgba');
+  static MimeType get jpeg => MimeType._('jpeg', 'image/jpeg');
+  static MimeType get png => MimeType._('png', 'image/png');
+  static MimeType get pcd => MimeType._('pcd', 'pointcloud/pcd');
+
+  /// An unsupported MimeType takes in the String representation of the mimetype that is not supported.
+  const MimeType.unsupported(this._name) : this._type = 'unsupported';
+
+  static MimeType fromString(String mimeType) => _map[mimeType] ?? MimeType.unsupported(mimeType);
+
+  static bool isSupported(String mimeType) {
+    return _map.containsKey(mimeType);
+  }
+
+  bool operator ==(covariant MimeType other) {
+    return _name == other._name;
+  }
+
+  int get hashCode => Object.hash(_type, _name);
+
+  img.Image? decode(List<int> bytes) {
+    img.Decoder? decoder;
+    switch (_type) {
+      case 'viamRgba':
+        decoder = _ViamRGBADecoder();
+        break;
+      case 'jpeg':
+        decoder = img.JpegDecoder();
+        break;
+      case 'png':
+        decoder = img.PngDecoder();
+        break;
+      case 'pcd':
+        decoder = null;
+        break;
+      case 'unsupported':
+        decoder = null;
+        break;
+    }
+    if (decoder == null) {
+      return null;
+    }
+    return decoder.decode(Uint8List.fromList(bytes));
+  }
+}
+
+class ViamImage {
+  /// THe mimetype of the image
+  final MimeType mimeType;
+
+  /// The raw bytes of the image
+  final List<int> raw;
+
+  bool _imageDecoded = false;
+  img.Image? _image;
+
+  /// The decoded image, if available. If the [MimeType] is not supported, this will be null.
+  img.Image? get image {
+    if (_imageDecoded) {
+      return _image;
+    }
+    _image = this.mimeType.decode(this.raw);
+    _imageDecoded = true;
```

Comment:
Should you check if _image is null before setting _imageDecoded? There seems to be no error handling if a new / unsupported mime type is passed.

Reply:
Perhaps `_imageDecoded` is a misnomer -- it's just a boolean to see if we've attempted decoding, not whether it was successful. Because if it was unsuccessful, we don't want to try decoding again (nothing will have changed, it'll be a waste of time). And if it was successful, we'll have the image. 

I can't simply do a `if (_image !== null)` test because it's perfectly OK for the `_image` to be `null`. So I have to check to see if we've attempted decoding already, and skip decoding a second time if yes